diag_manager/diag_output.F90

```diff
@@ -214,11 +216,37 @@ SUBROUTINE diag_output_init(file_name, FORMAT, file_title, file_unit,&
 
     !---- open output file (return file_unit id) -----
     IF ( domain .NE. NULL_DOMAIN2D ) THEN
+     !> Check if there is an io_domain
+     iF ( associated(mpp_get_io_domain(domain)) ) then
        fileob => fileobj
        if (.not.check_if_open(fileob)) call open_check(open_file(fileobj, trim(fname_no_tile)//".nc", "overwrite", &
                             domain, nc_format="64bit", is_restart=.false.))
        fnum_domain = "2d" ! 2d domain
        file_unit = 2
+     elSE !< No io domain, so every core is going to write its own file.
+       fileob => fileobjND
+       mype = mpp_pe()
+       write(mype_string,'(I0)') mype
+       select case (mype)
+          case ( : 9)
+               mype_string = "000"//trim(mype_string)
+          case (10 : 99)
+               mype_string = "00"//trim(mype_string)
+          case (100 : 999)
+               mype_string = "0"//trim(mype_string)
+          case (1000 : 9999)
+               mype_string = trim(mype_string)
+          case default
+               CALL error_mesg('diag_output_init', "The PE number is above 9999, so you can not add 4 digits to the "//&
+                    "end of the file name "//trim(file_name) ,FATAL)
+        end select
```

Comment:
This can be accomplished using a formatted `write` statement, similar to:

```Fortran
write (mype_string, '(I0.4)') mype
```

Reply:
Also, is there a check to make sure that the run is not using > 10,000 ranks?

Reply:
I increased the string length to 9, so it will be able to work if the PE number is greater than 9999.  It will write a minimum of 4 digits to `mype_string`.  @underwoo  and I tested this with gfortran and ifort.